### ISSUE TITLE
rxjs 6.3 mergeMap no longer works with Observable from a promise

### ISSUE THREAD

**Issue by menved** [ReactiveX/rxjs#4073]
rxjs version 6.3 mergeMap no longer works with an Observable from a promise (compared with version 6.2.2).
The Observable never subscribes.

```js
myObservable
.pipe(
  mergeMap(x =>
     from(Promise.resolve(['test']))
  )
.subscribe()

```
**Environment**
- Runtime: Node v10.9.0
- RxJS version: 6.3
- webpack 4.17.1

**Comment by cartant** [comment]
See #4071 and #4072. Hopefully, this is the same problem - which has been fixed.

**Comment by LM-G** [comment]
@cartant Hello, there is a problem on the `from`, if you do something like this

```javascript
from(SOME_PROMISE).pipe(switchMap(() => SOME_OBSERVABLE)).subscribe(...)
```
with SOME_OBSERVABLE getting an event in the future, the subscription of SOME_OBSERVABLE is canceled instantly.

**Comment by cartant** [comment]
@LM-G as stated above, see the failing tests that were fixed in #4072. Said tests used `from` with promises.

**Comment by menved** [comment]
Version 6.3.1 has solved the problem :)
